src/Vector.cpp

```diff
@@ -0,0 +1,215 @@
+#pragma once
+#include "Vector.h"
+
+
+Vector::Vector(const size_t& size, const int& value)
+    : _size(0), _capacity(0), _data(nullptr) {
+    for (size_t i = 0; i < size; i++) {
+        push_back(value);
```

Comment:
Populating the array like this means that you call the `push_back()` function with the `if` on EVERY iteration of the loop. This is inefficient for no good reason. Consider changing this implementation